app/src/main/java/com/cp/tasky/agenda/overview/domain/OverviewUseCase.kt

```diff
@@ -0,0 +1,68 @@
+package com.cp.tasky.agenda.overview.domain
+
+import com.cp.tasky.agenda.overview.domain.model.Overview
+import com.cp.tasky.agenda.shared.data.model.Event
+import com.cp.tasky.agenda.shared.data.model.Items
+import com.cp.tasky.agenda.shared.data.model.Reminder
+import com.cp.tasky.agenda.shared.data.model.Task
+
+object AgendaOverviewUseCase {
+
+    private const val DEFAULT_INITIALS = "NA"
+
+    /**
+     * All agenda items should be ordered
+     * ascending by time
+     */
+    fun sortAgendaItemsByAscendingTime(items: Overview): List<Items> {
+        val tempList = mutableListOf<Items>()
+        tempList.addAll(items.events)
+        tempList.addAll(items.tasks)
+        tempList.addAll(items.reminders)
+
+        return tempList.sortedBy { getRemindAt(it) }
+    }
+
+    private fun getRemindAt(item: Items): Long {
+        return when (item) {
+            is Event -> item.remindAt
+            is Task -> item.remindAt
+            is Reminder -> item.remindAt
+            else -> 0
+        }
```

Comment:
Given that you need to check each specific item type to access a common property of all agenda items, could there be a better way to model this, so you could just do `item.remindAt` regardless of its type?